### DISCUSSION TITLE
HttpRequest raw usage (recursive definition?)

### DISCUSSION THREAD

**Discussion by jlarroque** [Kong/unirest-java#446] [Q&A] [answered]
Hi there, I have a problem with how I use HttpRequest, Sonar always complains that I'm using it as raw:

![image](https://user-images.githubusercontent.com/291711/175530207-9a99a8f4-a4de-406a-824b-bd36e86ebe38.png)
![image](https://user-images.githubusercontent.com/291711/175530512-9b475b75-3629-4bc2-abf9-29fc1fd66764.png)

Can you point me to an example or explain to me how should I address that? I am really confused because it seems that I should parametrize it with a class that also extends from HttpRequest:

![image](https://user-images.githubusercontent.com/291711/175530922-1a7d9579-d0a8-4b05-812a-d33130ef30d9.png)

I'm using Unirest 3.13.6

**Comment by ryber** [comment, marked as answer]
yes, the type is recursive because it's a builder and as you get more specific there are instances of the builder that have new options (posts can have bodies, gets cant, multi-part bodies have options that single part's cant, etc)

There isn't anything wrong with how you are using it, Sonar is just a suggestion, it's not always right. But you can probably get it to shut up with ```HttpRequest<?> request = ...```
